examples/air-quality-sensor-app/linux/AirQualitySensorAppAttrUpdateDelegate.cpp

```diff
@@ -0,0 +1,190 @@
+std::map<std::string, ConcentrationFuncPointers> ConcentrationFuncMap = {
+    { "CarbonMonoxideConcentrationMeasurement",
+      { &CarbonMonoxideConcentrationMeasurement::Attributes::MeasuredValue::Set
+        /* Add required function maps */ } },
+    { "CarbonDioxideConcentrationMeasurement",
+      { &CarbonDioxideConcentrationMeasurement::Attributes::MeasuredValue::Set
+        /* Add required function maps */ } },
+    { "NitrogenDioxideConcentrationMeasurement",
+      { &NitrogenDioxideConcentrationMeasurement::Attributes::MeasuredValue::Set
+        /* Add required function maps */ } },
+    { "OzoneConcentrationMeasurement",
+      { &OzoneConcentrationMeasurement::Attributes::MeasuredValue::Set
+        /* Add required function maps */ } },
+    { "Pm25ConcentrationMeasurement",
+      { &Pm25ConcentrationMeasurement::Attributes::MeasuredValue::Set
+        /* Add required function maps */ } },
+    { "FormaldehydeConcentration",
+      { &FormaldehydeConcentrationMeasurement::Attributes::MeasuredValue::Set
+        /* Add required function maps */ } },
+    { "Pm1ConcentrationMeasurement",
+      { &Pm1ConcentrationMeasurement::Attributes::MeasuredValue::Set
+        /* Add required function maps */ } },
+    { "Pm10ConcentrationMeasurement",
+      { &Pm10ConcentrationMeasurement::Attributes::MeasuredValue::Set
+        /* Add required function maps */ } },
+    { "TotalVolatileOrganicCompoundsConcentrationMeasurement",
+      { &TotalVolatileOrganicCompoundsConcentrationMeasurement::Attributes::MeasuredValue::Set
+        /* Add required function maps */ } },
+    { "RadonConcentrationMeasurement",
+      { &RadonConcentrationMeasurement::Attributes::MeasuredValue::Set
+        /* Add required function maps */ } }
+};
+
+AirQualitySensorAttrUpdateHandler * AirQualitySensorAttrUpdateHandler::FromJSON(const char * json)
+{
+    Json::Reader reader;
+    Json::Value value;
+
+    if (!reader.parse(json, value))
+    {
+        ChipLogError(NotSpecified,
+                     "AllClusters App: Error parsing JSON with error %s:", reader.getFormattedErrorMessages().c_str());
+        return nullptr;
+    }
+
+    if (value.empty() || !value.isObject())
+    {
+        ChipLogError(NotSpecified, "AllClusters App: Invalid JSON command received");
+        return nullptr;
+    }
+
+    if (!value.isMember("Name") || !value["Name"].isString())
+    {
+        ChipLogError(NotSpecified, "AllClusters App: Invalid JSON command received: command name is missing");
+        return nullptr;
+    }
+
+    return Platform::New<AirQualitySensorAttrUpdateHandler>(std::move(value));
+}
+
+void AirQualitySensorAttrUpdateHandler::HandleCommand(intptr_t context)
+{
+    auto * self             = reinterpret_cast<AirQualitySensorAttrUpdateHandler *>(context);
+    std::string clusterName = self->mJsonValue["Name"].asString();
+
+    VerifyOrReturn(!self->mJsonValue.empty(), ChipLogError(NotSpecified, "Invalid JSON event command received"));
```

Comment:
On error return, this leaks, no?

Reply:
Fix in this PR:
https://github.com/project-chip/connectedhomeip/pull/28619

Reply:
Thank you!